### PR TITLE
Pow3r: stage 1

### PR DESCRIPTION
**Pull request by PJB3005**
https://hackmd.io/@ss14/lowpower

This initial PR will cover the basic power graph management and code cleanup. Future PRs will add stuff like lowpower effects, UIs for various machines, balancing, etc...

Other technical changes:
1. ECSify a ton of power code
2. just clean it all up in general
3. move stuff to the correct place related to power
4. ECS node groups.
5. node groups can now be visualized with the `nodevis` command.

:cl:
- tweak: Rewrote most of the power system.
- add: All power supplying objects now have a built-in ramping system. If a large amount of extra power is suddenly needed, engines and batteries on the grid need a bit of time to "ramp up" to match.
- add: Power is passed through batteries immediately. This means that APCs are basically on standby until engine power or such cuts out. When the engine cuts out, heavy electrical grids will be underpowered for a second or two until backup batteries can spin up to fill the gap.

### CODE REVIEW

**Comment by SweptWasTaken** [comment]
That "spin-up" after engine power being cut could have some pretty cool effect like lights flickering.

**Comment by PJB3005** [comment]
Yes, that is the point

**Comment by SweptWasTaken** [comment]
> 
> 
> Yes, that is the point

Did you even read my comment?

**Comment by PJB3005** [comment]
> 
> 
> > Yes, that is the point
> 
> Did you even read my comment?

Yes

**Comment by SweptWasTaken** [comment]
> 
> 
> > > Yes, that is the point
> > 
> > 
> > Did you even read my comment?
> 
> Yes

Ok

File: Content.IntegrationTests/Tests/Construction/ComputerIntegration.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                    FindComputer();
+
+                    Assert.That(serverComputer.Prototype?.ID ?? "", Is.EqualTo("ComputerId"));
+                });
+
+                await RunTicksSync(client, server, 1);
+            }
+        }
+    }
+}
```
Comment (Zumorica) [inline Content.IntegrationTests/Tests/Construction/ComputerIntegration.cs:95]: Oh hey, this feels familiar! :eyes: 
Did you commit it by mistake, or do you want to have it on the repo?

File: Content.Server/Power/Components/CableComponent.cs [hunk 2]
```diff
@@ -2,35 +2,35 @@
...
-    public class WireComponent : Component, IInteractUsing
+    public class CableComponent : Component, IInteractUsing
     {
-        public override string Name => "Wire";
+        public override string Name => "Cable";
 
         [ViewVariables]
-        [DataField("wireDroppedOnCutPrototype")]
+        [DataField("cableDroppedOnCutPrototype")]
         private string? _wireDroppedOnCutPrototype = "HVWireStack1";
```
Comment (Zumorica) [inline Content.Server/Power/Components/CableComponent.cs:24]: Could rename this field too, maybe

File: Content.Server/Power/Components/CablePlacerComponent.cs [hunk 3]
```diff
@@ -9,23 +10,23 @@
...
         [Dependency] private readonly IMapManager _mapManager = default!;
 
         /// <inheritdoc />
-        public override string Name => "WirePlacer";
+        public override string Name => "CablePlacer";
 
         [ViewVariables]
         [DataField("wirePrototypeID")]
-        private string? _wirePrototypeID = "HVWire";
+        private string? _wirePrototypeID = "HVCable";
```
Comment (Zumorica) [inline Content.Server/Power/Components/CablePlacerComponent.cs:25]: Rename these or nah?

File: Content.Server/Power/DrainAllBatteriesCommand.cs [hunk 4]
```diff
@@ -23,6 +23,12 @@ public void Execute(IConsoleShell shell, string argStr, string[] args)
                 return;
             }
 
+            var comp = IoCManager.Resolve<IComponentManager>();
+            foreach (var VARIABLE in comp.EntityQuery<BatteryComponent>())
+            {
+
+            }
```
Comment (Zumorica) [inline Content.Server/Power/DrainAllBatteriesCommand.cs:30]: Oopsie?

**Review by Zumorica — APPROVED** [review]
I skimmed through a bunch of things that I didn't understand, but the things that I did understand looked good. Good to merge aside from some minor nitpicks I pointed out

File: Content.IntegrationTests/Tests/Construction/ComputerIntegration.cs (hunk 1 above)
Comment (PJB3005) [reply Content.IntegrationTests/Tests/Construction/ComputerIntegration.cs:95]: Oops

**Comment by SweptWasTaken** [comment]
🥵🥺

**Pull request merged**